### PR TITLE
Separate SourcePawn specific code from the core

### LINKED ISSUE #50: Architecture changes
## Description

My suggestion is move binds for other languages to extentions, in core create only interfaces for another modules.
With this we can easier create binds for more languages(if someone will want add support for other language).

Question for discussion:
Maybe better develop metamod instead metaplugin for interfaces?

### PR DESCRIPTION
**Pull request by Amaroq7**
This is the attempt to ditch the SP code from the core of SPMod. The point of this PR is to track the current progress.

## Description
As discussed in #50. Closes #50.

## Motivation and Context
It'll help to implement other languages than just SourcePawn.

## How has this been tested?
Not ready to test yet.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/SrvCommand.cpp [hunk 1]
```diff
@@ -55,8 +50,8 @@ void SPModInfoCommand()
                                        std::setw(authWidth),
                                       "author",
                                       "filename");
-            std::size_t pos = 1;
-            for (auto entry : gSPGlobal->getPluginManagerCore()->getPluginsList())
+            //std::size_t pos = 1;
```
Comment (Mistrick) [inline src/SrvCommand.cpp]: Rewrite this for exts?

Comment (Amaroq7) [reply src/SrvCommand.cpp]: Forgot about this

Comment (Amaroq7) [reply src/SrvCommand.cpp]: Ready, but need to wait until https://github.com/dreamstalker/rehlds/pull/697 gets merged.

File: subprojects/example/ext.cpp [hunk 2]
```diff
@@ -15,20 +17,24 @@
...
+}
 
-    const char *getName() const override
-    {
-        return "spmod";
-    }
-};
+SPMOD_API bool SPMod_Init()
+{
+    return true;
```
Comment (Mistrick) [inline subprojects/example/ext.cpp]: add some example of using SPGlobal?

File: src/PlayerSystem.cpp [hunk 3]
```diff
@@ -230,12 +213,20 @@ bool PlayerMngr::ClientConnect(edict_t *pEntity,
...
-    // callback for modules
-    for (auto *listener : getListenerList())
-    {
-        if (!listener->OnClientConnect(pEntity, pszName, pszAddress, szRejectReason))
-            return false;
-    }
+    using def = ForwardMngr::FwdDefault;
+    using sf = IForward::StringFlags;
+
+    std::shared_ptr<Forward> fwdPlrConnect = gSPGlobal->getForwardManagerCore()->getDefaultForward(def::ClientConnect);
```
Comment (In-line) [inline src/PlayerSystem.cpp]: auto?

File: src/SrvCommand.cpp (hunk 1 above)
Comment (Mistrick) [reply src/SrvCommand.cpp]: RP merged

Comment (Amaroq7) [reply src/SrvCommand.cpp]: It's fixed in the current revision

**Pull request merged**